### PR TITLE
[Examples] Add GPTQ W4AFP8 Llama 3 8B example

### LINKED ISSUE #2236: Generate and validate W4AFp8 GPTQ Example
- Generate end-to-end Llama3 8b examples which apply int quantization to the weights and FP8 dynamic per token quantization to the activations using GPTQ.

- Validate the generated example in vLLM

See https://github.com/vllm-project/llm-compressor/issues/2235 for complete details

### PR DESCRIPTION
**Pull request by Etelis**
Add end-to-end example demonstrating W4AFP8 quantization (INT4 weights + FP8 dynamic activations) using GPTQ for Llama 3 8B.

## Test Plan

- [x] Verify the example runs successfully with the updated compressed-tensors
- [x] Validate the quantized checkpoint loads in vLLM
- [x] Run GSM8K evaluation on the quantized model

## Related Issues

Closes #2236
Relates to #2235

## Dependencies

Requires vllm-project/compressed-tensors#542 to be merged first (adds the `W4AFP8` preset scheme).

### CODE REVIEW

**Comment by Etelis** [comment]
> because this is a new scheme, we should create a new folder, not put it under quantization_w4a16

Can we make the CI tests run to see it actually passes?

**Comment by dsikka** [comment]
> > because this is a new scheme, we should create a new folder, not put it under quantization_w4a16
> 
> Can we make the CI tests run to see it actually passes?

The CI does not run example scripts. Were you able to generate a checkpoint and run the gsm8k eval?

**Comment by Etelis** [comment]
> > > because this is a new scheme, we should create a new folder, not put it under quantization_w4a16
> 
> > 
> 
> > Can we make the CI tests run to see it actually passes?
> 
> 
> 
> The CI does not run example scripts. Were you able to generate a checkpoint and run the gsm8k eval?

I need to wait for an approval on HF by Meta.
Will update once I got it

**Comment by Etelis** [comment]
@dsikka 

LOGS OF EXECUTION: 

[gptq_quantize_log.log](https://github.com/user-attachments/files/24717489/gptq_quantize_log.log)
[gptq_vllm_log.log](https://github.com/user-attachments/files/24717490/gptq_vllm_log.log)

@dsikka 
Ran it on an A100 (42GB) added the logs, 
Ran GS8MK with 50 samples (as more wouldn't be possible for me to run in a short period of time)


Machine:
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 570.86.10              Driver Version: 570.86.10      CUDA Version: 12.8     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA A100-PCIE-40GB          On  |   00000000:41:00.0 Off |                    0 |
| N/A   66C    P0            229W /  250W |    3523MiB /  40960MiB |     99%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
+-----------------------------------------------------------------------------------------+


=== OS ===
PRETTY_NAME="Ubuntu 24.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="24.04"
VERSION="24.04.3 LTS (Noble Numbat)"
VERSION_CODENAME=noble
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=noble
LOGO=ubuntu-logo

=== Kernel ===
Linux 1b4069f8d1f2 6.8.0-65-generic #68~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Jul 15 18:06:34 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

=== CPU ===
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        48 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               128

**Comment by Etelis** [comment]
> Hey! Thank you for running the eval! This lgtm overall. Do you mind sharing the checkpoint on the HF hub?
> 
> Thanks

https://huggingface.co/Etelis/Meta-Llama-3-8B-Instruct-W4AFP8-GPTQ/tree/main

**Comment by dsikka** [comment]
Hi @Etelis 
Were the evals generated using vLLM?
I dont believe this checkpoint will run in vLLM as the weights are not packed: https://github.com/vllm-project/vllm/blob/aa7f37ccfa168d40c89d1a104a870757666fcb56/vllm/model_executor/layers/quantization/compressed_tensors/schemes/compressed_tensors_w4a8_fp8.py#L118
You will need to update the example to provide the `quantization_format` when saving the model as `pack-quantized` https://github.com/vllm-project/compressed-tensors/blob/b055afcd104674cfc6ba79845ec505639c27d4fe/src/compressed_tensors/compressors/quantized_compressors/pack_quantized.py#L33

**Comment by Etelis** [comment]
> Hi @Etelis Were the evals generated using vLLM? I dont believe this checkpoint will run in vLLM as the weights are not packed: https://github.com/vllm-project/vllm/blob/aa7f37ccfa168d40c89d1a104a870757666fcb56/vllm/model_executor/layers/quantization/compressed_tensors/schemes/compressed_tensors_w4a8_fp8.py#L118 You will need to update the example to provide the `quantization_format` when saving the model as `pack-quantized` https://github.com/vllm-project/compressed-tensors/blob/b055afcd104674cfc6ba79845ec505639c27d4fe/src/compressed_tensors/compressors/quantized_compressors/pack_quantized.py#L33

Let me redo, yes I used VLLM with that checkpoint, I added the logs as well
I'll redo

**Comment by Etelis** [comment]
@dsikka 
`r, loading attributes (e.g. ['run_compressed']) will be overwritten with the one you passed to `from_pretrained`. The rest will be ignored.
  warnings.warn(warning_msg)
Compressing model: 224it [00:00, 2134.68it/s]
Loading checkpoint shards: 100%|█████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:02<00:00,  1.01s/it]
README.md: 7.93kB [00:00, 28.4MB/s]
main/train-00000-of-00001.parquet: 100%|██████████████████████████████████████████████████████████████████████████████| 2.31M/2.31M [00:00<00:00, 8.25MB/s]
main/test-00000-of-00001.parquet: 100%|█████████████████████████████████████████████████████████████████████████████████| 419k/419k [00:00<00:00, 4.37MB/s]
Generating train split: 100%|███████████████████████████████████████████████████████████████████████████████| 7473/7473 [00:00<00:00, 236079.46 examples/s]
Generating test split: 100%|████████████████████████████████████████████████████████████████████████████████| 1319/1319 [00:00<00:00, 183000.46 examples/s]
[2026-01-19 19:11:07] INFO __init__.py:695: Selected tasks:
[2026-01-19 19:11:07] INFO __init__.py:686: Task: gsm8k (gsm8k/gsm8k.yaml)
[2026-01-19 19:11:07] INFO evaluator.py:305: gsm8k: Using gen_kwargs: {'until': ['Question:', '</s>', '<|im_end|>'], 'do_sample': False, 'temperature': 0.0}
[2026-01-19 19:11:07] INFO task.py:434: Building contexts for gsm8k on rank 0...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:00<00:00, 344.45it/s]
[2026-01-19 19:11:07] INFO evaluator.py:574: Running generate_until requests
Running generate_until requests:   0%|                                                                                              | 0/50 [00:00<?, ?it/s]Passed argument batch_size = auto. Detecting largest batch size
Determined Largest batch size: 2
The following generation flags are not valid and may be ignored: ['temperature', 'top_p']. Set `TRANSFORMERS_VERBOSITY=info` for more details.
Running generate_until requests: 100%|█████████████████████████████████████████████████████████████████████████████████████| 50/50 [21:26<00:00, 25.73s/it]
[2026-01-19 19:32:40] INFO evaluation_tracker.py:280: Output path not provided, skipping saving results aggregated
hf (pretrained=./Meta-Llama-3-8B-Instruct-W4AFP8), gen_kwargs: (None), limit: 50.0, num_fewshot: None, batch_size: auto
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value|   |Stderr|
|-----|------:|----------------|-----:|-----------|---|----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  | 0.66|±  |0.0677|
|     |       |strict-match    |     5|exact_match|↑  | 0.66|±  |0.0677|
`

Done and updated

**Comment by dsikka** [comment]
Thank you! The PR is showing no line changes?

**Comment by Etelis** [comment]
> Thank you! The PR is showing no line changes?

Sorry I screwed it by mistake when trying to fix DCO
